src/styles/utils/_functions.scss

```diff
@@ -0,0 +1,12 @@
+@function max-width($cardsNumber, $cardWidth, $cardsMargin, $catalogPadding) {
+  $max-width: 4 * ($cardWidth)+ ($cardsMargin * 3) + (2 * $catalogPadding);
+
+  @if $cardsNumber/4 < 1 {
+    $max-width: $cardsNumber
+      * ($cardWidth)
+      + ($cardsMargin * ($cardsNumber - 1))
+      + (2 * $catalogPadding);
+  }
+
+  @return $max-width;
+}
```

Comment:
Can't understand what you calculate here. We have a requirement in the task 'there should always be 4 cards in a row (not 2, 3 or 5)'. The number of cards in the row changes in you case if we change the width of viewport.
Just give your catalog fixed width (4 cards + margins between them).

Reply:
// here calculated max-width if the number of cards is less than 4, 
13 [LAYOUT] - Use sass functions for dynamic max-width calculation, your catalog container should work fine for any number of cards.